Add option to new-note-namer, to underline the heading instead of highlighting it with #

Mystechry opened the pull request:
The script `new-note-namer` deviates from how QOwnNotes highlights the automatically created heading. Instead of underlining the first line with `======`, it precedes it with `#`.

I prefer the way how QOwnNotes highlights the first line, so I added a checkbox to `new-note-namer`'s settings that restores the default behavior.

I have done some testing and did not notice any regression. Please let me know, if anything looks suspicious.

I do not have any particular preference for which heading style shall be used as the default. I suggest to default to my style and underline the first line with  `=====`, so that it matches QOwnNotes default aesthetics. However, I do not have a strong opinion on this and respect the author's preference to default to `#`, so I use that for default. Feel free, to reject the PR, if I shall use `=====` as default.

pbek commented on new-note-namer/new-note-namer.qml:
```diff
@@ -41,7 +49,11 @@ QtObject {
+    if (underlineHeading){
+          return newName + "\n========================"
+    } else {
+          return "# " + newName;
```
Can you please adapt the amount of whitespace?

Mystechry replied:
I fixed my portion of the code. The original script is not properly consistent with its use of whitespace in the first place. Please let me know if I shall unify the whitespacing throughout the whole script.

pbek replied:
The whole script is a mess, you are right. Please do so in another PR.

pbek commented on new-note-namer/new-note-namer.qml:
```diff
@@ -41,7 +49,11 @@ QtObject {
     script.log(note.fileLastModified)
 
-    return "# " + newName;
+    if (underlineHeading){
```
Please use `) {`.

Mystechry replied:
Fixed

Mystechry commented on new-note-namer/info.json:
```diff
@@ -2,9 +2,9 @@
   "identifier": "new-note-namer",
   "script": "new-note-namer.qml",
-  "authors": ["@diegovskytl"],
+  "authors": ["@diegovskytl,@Mystechry"],
```
Fixed

Mystechry replied:
Fixed

Mystechry commented on new-note-namer/new-note-namer.qml:
```diff
@@ -41,7 +49,11 @@ QtObject {
 
-    return "# " + newName;
+    if (underlineHeading){
+          return newName + "\n========================"
```
That is a great idea. The updated file contains that feature.

Mystechry commented:
1. Both files got updated. All your suggestions and requests are in.
2. Fixed a bug I introduced that cuts the first two characters if underlining the heading was checked.

Let me know if there are any further issues or suggestions.

pbek commented:
Thank you!
